### DISCUSSION TITLE
CSS conflict: applies the same CSS properties as

### DISCUSSION THREAD

**Discussion by Mihai-github** [tailwindlabs/tailwindcss#5749] [Help] [answered]
Hi,

   I've started recently using tailwind and I am having an issue with the fact that I cannot use in code-editor the same property but with different values in different window sizes. What I am trying to say  can be seen in the following example:

```
<div className="text-left  lg: text-center">
          <p style={{ color: "red" }}>salut</p>
</div>
```
   In vs-code (my code editor) having the tailwind extension, and also tested on the official tailwind playground I am getting the following warning ** text-center' applies the same CSS property as 'text-left'.(cssConflict).**
If there is a white space between the media query selector (in our case lg:) the style text-left wont be applied for different window sizes, only the **text-center** will be used everywhere (makes sense if there is a problem to take the latest class).
But if there is no white space between **lg** and **text-center** like below:

```
<div className="text-left   lg:text-center">
          <p style={{ color: "red" }}>salut</p>
 </div>
```

It Will work just fine, BUT... if you would have a more complicated style from my testing the only way I've found is having everywhere you would like to use something different based on the window size that has been previously declared you must use the same "hack" like below:

```
<div className="text-left bg-gray-700  lg:text-center lg:bg-gray-500">
          <p style={{ color: "#fff" }}>salut</p>
</div>
```

I'm opened to any suggestion and I would also like, if possible, some clarification as to why this is not working in the dev stage (coding environment), but in the playground works having the same warning.
![image](https://user-images.githubusercontent.com/56633911/136692296-815c9a92-9bcb-4661-bb30-2b594e9439eb.png)

**Comment by simonswiss** [comment, marked as answer]
Hey @Mihai-github 👋

I hope you're enjoying Tailwind CSS so far!

When applying [responsive variants](https://tailwindcss.com/docs/responsive-design), there must be no space between the variant and the utility class.

`lg:text-center` is correct, `lg: text-center` (with the space) is not.

---

>  from my testing the only way I've found is having everywhere you would like to use something different based on the window size that has been previously declared you must use the same "hack" like below:

```html
<div className="text-left bg-gray-700  lg:text-center lg:bg-gray-500">
          <p style={{ color: "#fff" }}>salut</p>
</div>
```

This is not a hack, this _is_ how Tailwind responsive classes work! You define styles for small screens, and can alter those styles at specific breakpoints where needed, like you do here for the `lg` breakpoint 👍

---

> `text-center' applies the same CSS property as 'text-left'.(cssConflict).`

When you have a space between the `lg:` variant and the utility class, those are considered two separate classes (a space is a delimited between two classnames!). Therefore, that error message makes sense: it warns you that you're trying to apply both `text-left` and `text-center` to the same element.

---

>  I would also like, if possible, some clarification as to why this is not working in the dev stage (coding environment), but in the playground works having the same warning.

I am not sure what your code environment looks like, so it's kinda hard to understand this question, but perhaps my explanation above solves the problem for you?

**Reply by Mihai-github** [reply]
Hi, 
   First of all, thank you for your response, and yes I like using it. I used it in the past in a project and the whole idea it's nice and I like the fact that is different from other libraries that can be used for design purposes having a different way of doing things and sometimes using something let's say "different" make things more interesting and having so much control over the design for some project/peoples can be a good or bad thing depending on the project goal peoples exp maybe.
   What are you saying is correct, but you can also use "lg: text-center" and is also correct with a space in case that you don't have previously declared the same option before and it would work. Probably it acts as an overwrite in this case if you have redeclared the same class, but for a different responsive variant w/o any space between variant(ex: lg) and (:). 
Just wanted to point out this behavior because maybe other peoples would have the same problem let's say and they get stuck. Not sure if that's the intended behavior or is something different.

**Reply by NovusOpus** [reply]
> Hi, First of all, thank you for your response, and yes I like using it. I used it in the past in a project and the whole idea it's nice and I like the fact that is different from other libraries that can be used for design purposes having a different way of doing things and sometimes using something let's say "different" make things more interesting and having so much control over the design for some project/peoples can be a good or bad thing depending on the project goal peoples exp maybe. What are you saying is correct, but you can also use "lg: text-center" and is also correct with a space in case that you don't have previously declared the same option before and it would work. Probably it acts as an overwrite in this case if you have redeclared the same class, but for a different responsive variant w/o any space between variant(ex: lg) and (:). Just wanted to point out this behavior because maybe other peoples would have the same problem let's say and they get stuck. Not sure if that's the intended behavior or is something different.

I think you've got the wrong idea of the way it works,  `lg: text-center` with space is not valid,  as you can see in this [TailwindPlay](https://play.tailwindcss.com/2dMlnZqGQP), when you're declaring `lg:text-center`, that means the text is going to be in the center only after the `lg: or Large breakpoint`.
In the `sm:, md:, or any custom breakpoint before lg:` the text should align left by default and only move to the center after reaching the `lg breakpoint`,  by using: `lg: text-center` it's going to translate into just `text-center` because `lg: ` is going to be "ignored" since it's empty or null. that means the text is going to be in the center for each or any breakpoint.

**Reply by Mihai-github** [reply]
Got it so everything that you declare for a responsive variant must have the breakpoint variant before, like below:
```
<div class=" font-bold lg:text-right lg:bg-gray-800">
    <p style="color: red">salut</p>
</div>
```
Till reaches the lg breakpoint the bg will be white, but when it reached the breakpoint will be bg-gray-800. Got it thanks.

**Reply by NovusOpus** [reply]
@Mihai-github That's right, You can read the [Responsive variants link](https://tailwindcss.com/docs/responsive-design) that @simonswiss posted in his reply to better understand, Specially the [Mobile First Section](https://tailwindcss.com/docs/responsive-design#targeting-mobile-screens) since I think it's very important!

**Reply by Mihai-github** [reply]
@novuspace Thank you, I have read it, but for some kind of reason, I thought something different. 
Now all is clear. 

Hope this discussion will help others 👍

**Reply by simonswiss** [reply]
Discussions like these are great, because they always seem to end up helping someone else down the road, 6 months later 👍

**Reply by Mihai-github** [reply]
@simonswiss Indeed :D that's why I am so happy to see if someone when having an error and post on a community platform for developers and gets answers is a really nice feeling to have in my option. Also down on the road, someone else could benefit from a well-done discussion with answers.

**Reply by feapolina** [reply]
Thank you so much, still helping people in 2024. **Obrigado!**

**Reply by realihorrud** [reply]
And still...

**Comment by curiouscod3** [comment]
I had to change my tailwindCSS intellisense classRegex option for a special case.
It causes the 'cssConflict' warning which is annoying !

```
 // https://github.com/tailwindlabs/tailwindcss-intellisense/issues/129
  "tailwindCSS.experimental.classRegex": [
    "`([^`]*)", // This can cause wrong 'cssConflict' issue
    "tw`([^`]*)",
    "classnames\\(([^)]*)\\)",
    "'([^']*)'", // This can cause wrong 'cssConflict' issue
    // "\"([^']*)\"", // This can cause wrong 'cssConflict' issue
  ],
```

For me, To fix this issue, the following option worked ! 
[TailwindCSS Lint Option](https://github.com/tailwindlabs/tailwindcss-intellisense#tailwindcssvalidate)

`"tailwindCSS.lint.cssConflict" : "ignore" `

This is my full TailwindCSS settings in the settings.json !

```
// For Auto Completion, Check "editor.quickSuggestions"
  "tailwindCSS.includeLanguages": {
    "plaintext": "html",
    "javascript": "javascriptreact",
    "typescript": "typescriptreact", // or javascriptreact
    "html": "HTML"
  },
  // https://github.com/tailwindlabs/tailwindcss-intellisense/issues/129
  "tailwindCSS.experimental.classRegex": [
    "`([^`]*)", // This can cause wrong 'cssConflict' issue
    "tw`([^`]*)",
    "classnames\\(([^)]*)\\)",
    "'([^']*)'", // This can cause wrong 'cssConflict' issue
    // "\"([^']*)\"", // This can cause wrong 'cssConflict' issue
  ],
  "tailwindCSS.emmetCompletions": false,
  "tailwindCSS.classAttributes": ["class, className, ngClass"],
  "tailwindcss.colorDecorators": "inherit", // inherit from "editor.colorDecorators"
  "tailwindcss.references": true,
  // "tailwindcss.diagnostics": {
  //   "enabled": true,
  //   "conflict": "strict",
  //   "emptyClass": true,
  //   "emptyGroup": true,
  //   "emptyCssProperty": true
  // },
  "tailwindcss.preferVariantWithParentheses": false,
  "tailwindcss.fallbackDefaultConfig": true,
  "tailwindcss.enabled": true,
  "tailwindcss.jsxPropImportChecking": true,
  "tailwindcss.rootFontSize": 16,
  "tailwindcss.logLevel": "info",
  "tailwindcss.documentColors": false,
  "tailwindcss.hoverColorHint": "none",
  // [tailwindCSS.lint Options] 
  // https://github.com/tailwindlabs/tailwindcss-intellisense#tailwindcssvalidate
  "tailwindCSS.lint.cssConflict" : "ignore",
```

**Comment by StyxUA** [comment]
Weird thing here: `border-white/75` and `active:border-stone-500` apparently apply the same CSS.

<img width="913" height="187" alt="Screenshot 2026-08-21 at 21 28 24" src="https://github.com/user-attachments/assets/76c2a4b9-e665-477e-9493-420f5a33d5fe" />

<img width="795" height="187" alt="Screenshot 2026-08-21 at 21 30 12" src="https://github.com/user-attachments/assets/a3d6742b-47af-4b4c-979b-05db167e649a" />


Any ideas, why is that so?

**Reply by RobinMalfait** [reply]
Can you double check if the vscode extension is up to date?

**Reply by StyxUA** [reply]
@RobinMalfait Sorry, should have clarified it's a PhpStorm. tailwindcss 4.3.3, @tailwind/language-server 0.14.29.

**Reply by StyxUA** [reply]
Same thing with `focus:`, but `hover:` is good. `text-...` also affected.

**Reply by RobinMalfait** [reply]
Aha I see, yeah the underlying language server needs updating. So if you can bump `@tailwindcss/language-server` to 0.16.0 at the time of writing this, then that should solve this.

**Reply by StyxUA** [reply]
Yes, that solved the issue. Thanks a lot!
